client/signup/controller.js

```diff
@@ -305,6 +305,16 @@ export default {
 		const isManageSiteFlow =
 			! excludeFromManageSiteFlows && ! isAddNewSiteFlow && isReEnteringSignupViaBrowserBack;
 
+		// If the user entered the signup flow from the checkout page, submit the domains step
+		// to allow the user to return to the plans page.
+		if ( isManageSiteFlow ) {
```

Comment:
Have you tried checking for `isReEnteringSignupViaBrowserBack` only here? It seems to be working well, but I might be missing some specific cases related to values of `! excludeFromManageSiteFlows` and `! isAddNewSiteFlow` being `true`. 🤔

Reply:
`! excludeFromManageSiteFlows` excludes DIFM flows ([added here](https://github.com/Automattic/wp-calypso/pull/58951)). I'm not sure if that's still needed, but it seems safer to keep it since it matches the current `isManageSiteFlow` behavior, like showing the domain skip button.

`! isAddNewSiteFlow` was [added here](https://github.com/Automattic/wp-calypso/pull/59010). It's checking for a ref param in the URL, because that would indicate starting the flow from another page, like `/start/domains?source=sites-dashboard&ref=topbar`.